### ISSUE TITLE
Control characters in strings should be escaped

### ISSUE THREAD

**Issue by travisbrown** [circe/circe-yaml#19]
This issue is a restatement and minimization of #15. The problem is that control characters in strings aren't being escaped during printing:

```scala
scala> val Right(json) = io.circe.jawn.parse("\"\\u0085\"")
json: io.circe.Json = "\u0085"

scala> io.circe.yaml.Printer.spaces2.pretty(json)
res0: String =
"'  '
"
```

From [the spec](http://yaml.org/spec/1.1/#id872840):

> YAML streams use the _printable_ subset of the Unicode character set. On input, a YAML processor must accept all printable ASCII characters, the space, tab, line break, and all Unicode characters beyond `#x9F`. On output, a YAML processor must only produce these acceptable characters, and should also escape all non-printable Unicode characters. The allowed character range explicitly excludes the surrogate block `#xD800-#xDFFF`, `DEL #x7F`, the C0 control block `#x0-#x1F` (except for `#x9`, `#xA`, and `#xD`), the C1 control block `#x80-#x9F`, `#xFFFE`, and `#xFFFF`. Any such characters must be presented using escape sequences.

I have no idea how YAML libraries in general (or SnakeYAML in particular) handle this issue. It seems reasonable to me to say that if a string contains these characters, it must be represented using the double-quoted style no matter what the user-specified scalar style is, but I'd prefer for someone who actually knows something about YAML to say whether that's what we should do here.

**Comment by travisbrown** [comment]
Note that this issue is currently causing intermittent test failures, but the PR that fixes it should include a test case that consistently fails without the change.

**Comment by non** [comment]
I have some additional findings.

I exhaustively tested individual characters and identified two that are giving us problems: 0x0085 (as above) and 0xFEFF (zero-width no break space, which is lost in translation). These both seem like issues 

I *think* this is a bug in SnakeYAML, although I can't 100% confirm that, since the API seems confusing. But it seems like the `processScalar` analysis in https://bitbucket.org/asomov/snakeyaml/src/04378d05777d21d114a9cdc24976ad49c8919222/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java?at=default&fileviewer=file-view-default#Emitter.java-789 is not working correctly.

**Comment by travisbrown** [comment]
Resolved in #29. Thanks for investigating, @non!
